zerver/tests/test_home.py

```diff
@@ -32,6 +32,7 @@
 from zerver.views.home import compute_navbar_logo_url
 from zerver.worker.queue_processors import UserActivityWorker
 
+logger_string = "zulip.soft_deactivation"
```

Comment:
There are a few other places this has been declared and re-used effectively to assert log output strings. It is great but not sure if it's a good idea to declare globally.
- - -
I wonder if it's a good idea to add a `logger_string=root`(which can be overridden in the subclass accordingly) and a function like this in `ZulipTestCase`
```
def logger_output(self, output_string: str, type: str) -> str:
        return f'{type.upper()}:zulip.auth.{self.backend.name}:{output_string}'
```
This way we can hide the `INFO:{logger_string}`(which kind of feels like noise to me) from the expected log output.
For example the following
```
       self.assertEqual(info_log.output, [
            f'INFO:{logger_string}:Soft Deactivated user {long_term_idle_user.id}',
            f'INFO:{logger_string}:Soft-deactivated batch of 1 users; 0 remain to process'
        ])
```
can be written as
```
       self.assertEqual(info_log.output, [
            self.log_output(f'Soft Deactivated user {long_term_idle_user.id}', 'info'),
            self.log_output(f'Soft-deactivated batch of 1 users; 0 remain to process', 'info')
        ])
```
or maybe we could simplify that to avoid writing `self.log_output` by having some function, say `expect_logs` which could look something like this
```
def expect_logs(self, outputs: List[Tuple[str, str]]):
    expected = []
    for expected_log in outputs:
        output = expected_log[0]
        level = expected_log[1]
        expected.append(self.logger_output(output, level))
    return expected
```
and the same example can be simply written as
```
       self.assertEqual(info_log.output, self.expected_logs([
            (f'Soft Deactivated user {long_term_idle_user.id}', 'info'),
            (f'Soft-deactivated batch of 1 users; 0 remain to process', 'info')
        ])
```
@timabbott @mateuszmandera @showell Could you share your thoughts on this?